File: python/树/951. 翻转等价二叉树.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def flipEquiv(self, root1: TreeNode, root2: TreeNode) -> bool:
        if root1.val != root2.val:
            return False
        if root1.left.val != root2.left.val or root1.right.val != root2.right.val:
            root1.left,root1.right = root1.right,root1.left
        left = self.flipEquiv(root1.left,root2.left)
        right = self.flipEquiv(root2.right,root2.right)
        return left and right

File: go/tree/951. 翻转等价二叉树.go
package tree

func flipEquiv(root1 *TreeNode, root2 *TreeNode) bool {
	if root1 == root2 {
		return true
	}
	if root1 == nil || root2 == nil || root1.Val != root2.Val {
		return false
	}
	one := flipEquiv(root1.Left, root2.Left) && flipEquiv(root1.Right, root2.Right)
	two := flipEquiv(root1.Left, root2.Right) && flipEquiv(root1.Right, root2.Left)
	return one || two
}
